vertx-pojo-mapper-common/src/main/java/de/braintags/vertx/jomnigate/dataaccess/query/impl/Query.java

```diff
@@ -72,15 +73,60 @@ public final void execute(IFieldValueResolver resolver, int limit, int offset,
       if (syncResult.failed()) {
         resultHandler.handle(Future.failedFuture(syncResult.cause()));
       } else {
-        buildQueryExpression(resolver, result -> {
-          if (result.failed()) {
-            resultHandler.handle(Future.failedFuture(result.cause()));
-          } else {
-            IQueryExpression queryExpression = result.result();
-            queryExpression.setLimit(limit, offset);
-            internalExecute(queryExpression, resultHandler);
-          }
-        });
+        try {
+          Future<IQueryResult<T>> rf = Future.future();
+          rf.setHandler(resultHandler);
+          IObserverContext context = IObserverContext.createInstance();
+          preQuery(context).compose(pre -> executeQuery(resolver, limit, offset))
+              .compose(wr -> postQuery(wr, context, rf), rf);
+        } catch (Exception e) {
+          resultHandler.handle(Future.failedFuture(e));
+        }
```

Comment:
yes, definitely, cause the system is calling ObserverHandler.handleBeforeLoad / handleAfterLoad, for instance, which is working a while until its landing inside a handler. Minimum a ClassNotFoundException can occur because of configuration error. See TestObserverMapping